src/lib/operators/jit_operator/operators/jit_expression.hpp

```diff
@@ -46,18 +46,19 @@ class JitExpression {
   // Create a value JitExpression which returns the stored variant value
   explicit JitExpression(const JitTupleEntry& tuple_entry, const AllTypeVariant& variant);
   // Create a unary JitExpression
-  JitExpression(const std::shared_ptr<const JitExpression>& child, const JitExpressionType expression_type,
+  JitExpression(const std::shared_ptr<JitExpression>& child, const JitExpressionType expression_type,
                 const size_t result_tuple_index);
   // Create a binary JitExpression
-  JitExpression(const std::shared_ptr<const JitExpression>& left_child, const JitExpressionType expression_type,
-                const std::shared_ptr<const JitExpression>& right_child, const size_t result_tuple_index);
+  JitExpression(const std::shared_ptr<JitExpression>& left_child, const JitExpressionType expression_type,
+                const std::shared_ptr<JitExpression>& right_child, const size_t result_tuple_index);
 
   std::string to_string() const;
 
   JitExpressionType expression_type() const { return _expression_type; }
-  std::shared_ptr<const JitExpression> left_child() const { return _left_child; }
-  std::shared_ptr<const JitExpression> right_child() const { return _right_child; }
+  std::shared_ptr<JitExpression> left_child() const { return _left_child; }
+  std::shared_ptr<JitExpression> right_child() const { return _right_child; }
   const JitTupleEntry& result_entry() const { return _result_entry; }
+  JitTupleEntry& result_entry() { return _result_entry; }
```

Comment:
I would suggest to make all Jit class members which have getters public and remove their getters and setters. The const field still ensures that not all members can be modified. However, I did not change this in this PR to limit its size.

Reply:
Sounds good